canonical-web-and-design/maas-ui
# Build generic ActionForm component for use in machine and KVM lists.

## Done

- Built ActionForm component, which standardises common elements in forms that appear in the header strip (labelling, processing, styling).
- Changed ActionForm name in machine list to FieldlessForm.
- Originally I was going to convert all the touched files to TS but that would have made this PR huge. So instead I just converted all the index files (and the ActionForm) to TS. >_>
- Added selectors to get machines that are currently processing, and machines that are both selected and processing. This removes the need to pass `processing` and `setProcessing` props to form components.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/machines and check that action forms still work as expected.
  - Select one machine and perform an action
  - Select several machines and perform an action
  - Select several machines and select an action that not all machines can perform
  - Perform an action that will error (e.g. Set pool > Create pool > Enter a name that already exists)
- Go to /MAAS/r/kvm and check that the action forms work there too.

## Fixes

Fixes #1279 

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
